src/noise.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{FrequencyHz, Signal, TimeSecs};
+
+pub fn sample(rate: FrequencyHz, s: Signal) -> impl Iterator<Item = f64> {
+    (0..).map(move |n: u32| {
+        let sample_period = rate.period();
+        let t = n as f64 * sample_period.0;
+        s.at(TimeSecs(t))
+    })
+}
+
+/// Estimates the root mean square (RMS) of a signal over a period
+///
+/// # Arguments
+///
+/// * `s` - The `Signal` used to compute the RMS
+/// * `period` - The period (starting at phase=0) over which the computation is made
+///
+pub fn rms(s: Signal, period: TimeSecs) -> f64 {
+    // The number of samples should depend on the period.
+    let samples = 100;
+    let rate = FrequencyHz(((samples as f64) / period.0) as u32);
+
+    // The fold function is derived from:
+    //
+    // let a_1, ... a_n be the items of the iterator.
```

Comment:
I suggest rewording this comment or tweaking the pseudocode so it matches.

The items in the iterator iuc are of a tuple type (`(usize, f64)`), but in the pseudocode `a_n` is scalar.

Reply:
In the end I think the mix of maths and pseudo-code is too confusion so I decided to remove it.